### PR TITLE
feature: Add some form testing helpers

### PR DESCRIPTION
**Pull request by craigkuhns**
The new table and action testing helpers are great. The current project I'm working on has some interactive forms with disabled/enabled fields and hidden/visible fields depending on who is logged in or the state of other fields on the form. I wanted an easy way to test them so I took inspiration from the table testing helpers and I added the following helpers:

- assertFormExists: I haven't used this outside of use in the other testing helpers
- assertFormFieldExists: I haven't used this outside of use in the other testing helpers
- assertFormFieldIsDisabled
- assertFormFieldIsEnabled:
- assertFormFieldIsHidden
- assertFormFieldIsVisible

You can look at tests/src/Forms/FormsTest.php to see examples of how these helpers can be used. I also added some documentation showing simple use cases of how they work.

Regarding tests, I wasn't sure of the best place to add them so I put them all in FormsTest.php. Most of the helpers are actually about form fields, but they didn't seem to fit in the FieldTest.php file. I can move them around though if needed.

Let me know if there are any changes needed or if this would be better off as a plugin.

### CODE REVIEW

File: packages/forms/docs/07-testing.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+Or to ensure that a field is hidden you can pass the name to `assertFormFieldIsHidden`:
+
+```php
+use function Pest\Livewire\livewire;
+
+test('title is hidden', function () {
+    livewire(CreatePost::class)
+        ->assertFormFieldIsVisible('title');
```
Comment (rschoonheim) [inline packages/forms/docs/07-testing.md:59]: Is this correct? I assume you want to show the working of `assertFormFieldIsHidden('title');`

Comment (craigkuhns) [reply packages/forms/docs/07-testing.md:59]: Thanks, you're right!

File: packages/forms/src/Testing/TestsForms.php [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            $this->assertFormExists($formName);
+
+            $livewire = $this->instance();
+            $livewireClass = $livewire::class;
+
+            /** @var ComponentContainer $form */
+            $form = $livewire->$formName;
+
+            /** @var ?Field $field */
+            $field = data_get($form->getFlatFields(true), $fieldName, null);
```
Comment (danharrin) [inline packages/forms/src/Testing/TestsForms.php:51]: Please used a named parameter for `getFlatFields(true)`

Comment (danharrin) [reply packages/forms/src/Testing/TestsForms.php:51]: Also this is used in other places, update those also

File: packages/forms/src/Testing/TestsForms.php [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            $form = $livewire->$formName;
+
+            $fields = $form->getFlatFields(true);
+
+            $fields = $form->getFlatFields();
+
+            Assert::assertArrayHasKey(
+                $fieldName,
+                $fields,
+                "Failed asserting that a field with the name [{$fieldName}] is hidden on the form named [{$formName}] on the [{$livewireClass}] component."
```
Comment (danharrin) [inline packages/forms/src/Testing/TestsForms.php:154]:
```suggestion
                "Failed asserting that a field with the name [{$fieldName}] is visible on the form named [{$formName}] on the [{$livewireClass}] component."
```

File: packages/forms/src/Testing/TestsForms.php [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            /** @phpstan-ignore-next-line  */
+            $this->assertFormFieldExists($fieldName, $formName);
+
+            $livewire = $this->instance();
+            $livewireClass = $livewire::class;
+
+            /** @var ComponentContainer $form */
+            $form = $livewire->$formName;
+
+            $fields = $form->getFlatFields();
```
Comment (danharrin) [inline packages/forms/src/Testing/TestsForms.php:123]: Please explicitly pass `false` here instead of relying on the default, also using a named parameter

**Review by danharrin — CHANGES_REQUESTED** [review]
Also, please add a `.stubs.php` file for the forms package, like we have for Admin and Tables. This provides IDE autocompletion for the TestableLivewire class.

Apart from that, very nice job! I like it.

File: packages/forms/src/Testing/TestsForms.php (hunk 4 above)
Comment (craigkuhns) [reply packages/forms/src/Testing/TestsForms.php]: This is updated now

File: packages/forms/src/Testing/TestsForms.php (hunk 2 above)
Comment (craigkuhns) [reply packages/forms/src/Testing/TestsForms.php]: All of these are updated now

**Comment by craigkuhns** [comment]
Thanks @danharrin ! I've added the .stubs.php file and made the other suggested changes. Thanks for all of your work on Filament, it's the best admin panel I've used.

**Comment by danharrin** [comment]
Thank you very much for the kind words and the contribution

**Pull request merged**